modin/core/dataframe/pandas/dataframe/dataframe.py

```diff
@@ -2372,17 +2387,15 @@ def _copartition(self, axis, other, how, sort, force_repartition=False):
         Returns
         -------
         tuple
-            Tuple of (left data, right data list, joined index).
+            Tuple containing:
+                1) 2-d NumPy array of aligned left partitions
+                2) list of 2-d NumPy arrays of aligned right partitions
+                3) joined index along ``axis``
+                4) (optional) sizes of partitions along axis that partitioning was done on
```

Comment:
As I see we always return the sizes: either non empty list or an empty list. Optional seems to me we may return them or not return. Maybe it is worth elaborating that a bit.